annotations_models/train_random_forest_classifier_model_coepelduynen.ipynb

```diff
@@ -0,0 +1,1280 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Train Pixel Level Annotation Model\n",
+    "\n",
+    "### This notebook uses pixel level annotations to train a Random Forest Classifier to predict labels\n",
+    "\n",
+    "We assume Pixel level annotations are available, as produced by the \"../data/annotations/transform_polygon_annotations_to_pixels.ipynb\" notebook. Feature selection and grid_search for optimal parameters has been done in a separate notebook (\"Coepelduynen/make_train_model_on_annotations_coepelduynen.ipynb\") and those outcomes are taken as given in this notebook.\n",
+    "\n",
+    "Change the set Variables cell below as desired and then run the entire notebook to get cross_validation results as well as a final model trained on all data.\n",
+    "\n",
+    "Date: 2024-01-12\\\n",
+    "Author: Pieter Kouyzer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload 2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2024/02/01 17:05:45 WARNING mlflow.utils.autologging_utils: You are using an unsupported version of sklearn. If you encounter errors during autologging, try upgrading / downgrading sklearn to a supported version, or try upgrading MLflow.\n"
+     ]
+    }
+   ],
+   "source": [
+    "%matplotlib notebook\n",
+    "import pandas as pd\n",
+    "import pickle\n",
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "import pprint\n",
+    "\n",
+    "from training.train import train_imbalanced_model, cross_validation_balance_on_date\n",
+    "from training.utils import get_cross_validation_results_filepath, get_model_filepath\n",
+    "from training.metric_calculation import calculate_average_metrics, get_metrics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set Variables\n",
+    "location = \"Coepelduynen\"\n",
+    "satellite_constellation = \"PNEO\"\n",
+    "annotated_pixels_filepath = \"C:/repos/satellite-images-nso-datascience/data/annotations/annotations_pixel_dataframes/annotaties_coepelduynen_to_pixel_2023_scaled.parquet\""
```

Comment:
Perhaps redundant, but the idea behind having the scaler in this notebook instead of separately, means you shouldn't use the 'scaled' parquet files, but the original (unscaled) annotations parquet files instead. For the future.

Reply:
Moved to one notebook now